packages/frontend/src/blockchain/hooks/useBid.ts

```diff
@@ -0,0 +1,30 @@
+import { useChainId, useReadContract } from 'wagmi'
+import { AUCTION_ABI } from '@/blockchain/abi/auction'
+import { AUCTION_ADDRESSES } from '@/blockchain/auctionAddresses'
+import { Hex } from 'viem'
+
+export const useBid = (address: Hex | undefined) => {
```

Comment:
As the hook isn't used anywhere else I would merge `useReadContract` call with `useUserBid` function